cmd/common/signer/signer_test.go

```diff
@@ -34,6 +38,59 @@ func TestSigner(t *testing.T) {
 	ecdsa.Verify(&signer.key.PublicKey, util.ComputeSHA256(msg), r, s)
 }
 
+func TestSignerDifferentFormats(t *testing.T) {
+	key := `-----BEGIN EC PRIVATE KEY-----
+MHcCAQEEIOwCtOQIkowasuWoDQpXHgC547VHq+aBFaSyPOoV8mnGoAoGCCqGSM49
+AwEHoUQDQgAEEsrroAkPez9reWvJukufUqyfouJjakrKuhNBYuclkldqsLZ/TO+w
+ZsQXrlIqlmNalfYPX+NDDELqlpXQBeEqnA==
+-----END EC PRIVATE KEY-----`
+
+	pemBlock, _ := pem.Decode([]byte(key))
+	assert.NotNil(t, pemBlock)
+
+	ecPK, err := x509.ParseECPrivateKey(pemBlock.Bytes)
+	assert.NoError(t, err)
+
+	ec1, err := x509.MarshalECPrivateKey(ecPK)
+	assert.NoError(t, err)
+
+	pkcs8, err := x509.MarshalPKCS8PrivateKey(ecPK)
+	assert.NoError(t, err)
+
+	for _, testCase := range []struct {
+		description string
+		keyBytes    []byte
+	}{
+		{
+			description: "EC1",
+			keyBytes:    pem.EncodeToMemory(&pem.Block{Type: "EC Private Key", Bytes: ec1}),
+		},
+		{
+			description: "PKCS8",
+			keyBytes:    pem.EncodeToMemory(&pem.Block{Type: "EC Private Key", Bytes: pkcs8}),
```

Comment:
IIUC, the PEM block should simply be “PRIVATE KEY” for PKCS#8. I don’t know if it matters in practice.